engine/src/main/java/org/camunda/bpm/engine/impl/db/EnginePersistenceLogger.java

```diff
@@ -115,19 +113,21 @@ public ProcessEngineException alreadyMarkedEntityInEntityCacheException(String i
     ));
   }
 
-  public ProcessEngineException flushDbOperationException(List<DbOperation> operationsToFlush, DbOperation operation,
-      Throwable cause) {
+  public ProcessEngineException flushDbOperationException(List<DbOperation> operationsToFlush,
+                                                          DbOperation failedOperation,
+                                                          String message,
+                                                          Throwable e) {
 
-    String message = ExceptionUtil.collectExceptionMessages(cause);
     String exceptionMessage = exceptionMessage(
-      "004",
-      "Exception while executing Database Operation '{}' with message '{}'. Flush summary: \n {}",
-      operation.toString(),
-      message,
-      buildStringFromList(operationsToFlush)
+        "004",
+        "Exception while executing Database Operation '{}' with message '{}'. Flush summary: \n {}",
+        failedOperation,
+        message,
+        buildStringFromList(operationsToFlush)
```

Comment:
This will now only list a chunk of the flush (see the partitioning of operations in `DbEntityManager#flushDbOperationManager`). For debugging purposes, we should still include the entire flush in the exception message.

Reply:
👍